Source/WebCore/style/ElementRuleCollector.cpp

```diff
@@ -677,10 +680,34 @@ std::pair<bool, std::optional<Vector<ElementRuleCollector::ScopingRootWithDistan
             if (scopingRoots.isEmpty())
                 return false;
         } else {
-            // FIXME: the scoping root is the parent element of the owner node of the stylesheet where the @scope rule is defined. (If no such element exists, then the scoping root is the root of the containing node tree.
-            // We don't support those @scope rules without scope start yet
-            // https://bugs.webkit.org/show_bug.cgi?id=266399
-            return false;
+            // The scoping root is the parent element of the owner node of the stylesheet where the @scope rule is defined. (If no such element exists, then the scoping root is the root of the containing node tree.
+            RefPtr<const StyleSheetContents> styleSheetContents = rule->styleSheetContents();
+            if (!styleSheetContents)
+                return false;
+
+            auto appendNode= [&](const auto* client) {
+                const auto* owner = client->ownerNode();
```

Comment:
You could check the the node is actually in the current document (by comparing document() ptr) and ignore those that aren't.